src/Build/Instance/ProjectInstance.cs

```diff
@@ -2710,5 +2771,30 @@ private void CreatePropertiesSnapshot(Evaluation.Project.Data data, bool isImmut
                 _properties.Set(instance);
             }
         }
+
+        /// <summary>
+        /// Tracks this property read if it's from an environment variable or uninitialized.
+        /// </summary>
+        /// <param name="name">The name of the property being read.</param>
+        private void TrackPropertyRead(string name, bool propertyMissing)
```

Comment:
>propertyMissing [](start = 57, length = 15)

nit: 'propertyUninitialized' sounds more in line with what you called it above

Reply:
Updated it.